docs/_source/docs/architecture.rst

```diff
@@ -135,15 +135,222 @@ to demonstrate ways of organising different projects.
+"Indigo." Here are a few basic assumptions of this example:
+
+* Indigo's infrastructure will be deployed automatically using a CI/CD system (such as Jenkins).
+* The AWS role held by the CI/CD system cannot have sweeping, admin-level IAM permissions. It is
+  used by other teams in the company and cannot have its permissions modified for the Indigo
+  project.
+* Not all infrastructure in the organization is deployed/maintained by Sceptre and Indigo will need
+  to interact with that infrastructure.
+* All infrastructure for Indigo should be managed by Sceptre, including any required deployment
+  resources.
+* Developers will need to be able to stand up their own, isolated environments of Indigo that do not
+  conflict with production. These environments will have some different configurations to make them
+  cheaper for the company, but otherwise be identical to production.
+
+Project Structure
+*****************
+
+.. code-block:: text
+
+    config/
+        config.yaml             # top-level project configuration
+        indigo/                 # StackGroup for all Indigo's resources
+            config.yaml         # StackGroup Config for Indigo
+            application.yaml    # Stack Config for Indigo's application stack
+        project/                # StackGroup for resources needed to deploy Indigo
+            deployment.yaml     # Stack Config for Indigo's deployment resources
+    templates/
+        indigo/
+            application.yaml
+        project/
+            deployment.yaml
+    vars/
+        production.yaml         # var-file with the production environment's configurations
+        development.yaml        # var-file with configurations for development environments
+
+Example Configs
+***************
+
+Top-level project configuration
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. code-block:: yaml
+
+    # config/config.yaml
+
+    # This is the top-level config file for the whole project. Every stack in the project will
+    # inherit these settings unless they explicitly override them. At minimum, you need the
+    # project_code and region, but you can set whatever StackGroup and Stack configurations you
+    # want inherited by the rest of the project.
+
+    # By making the project-code parameterized by environment name, it guarantees that one
+    # environment's stacks do not conflict with another environment's stacks.
+    project_code: indigo-{{ var.environment }}
+    region: {{ var.region }}
+
+    # Setting the template_bucket_name with a !stack_output means that every stack in the project
+    # (except the deployment stack) will have its template uploaded to that bucket. It ALSO means
+    # that every stack in the project will inherit a dependency on the deployment stack (except the
+    # deployment stack... see its comments below)
+    template_bucket_name: !stack_output project/deployment.yaml::BucketName
```

Comment:
BucketName isn't defined in the example project/deployment.yaml so how does this dependency work?

Reply:
It's a `!stack_output` resolver, so it's referencing an output on the _template_ of that stack. Remember, outputs are not defined on stack configs; they're on the templates _referenced_ by stack configs.